src/sliding_window_split.cpp

```diff
@@ -0,0 +1,61 @@
+#include <sliding_window/shared.hpp>
+#include <sliding_window/split/reference_metadata.hpp>
+#include <sliding_window/split/reference_segments.hpp>
+
+namespace sliding_window::app
+{
+
+//-----------------------------
+//
+// Divide reference database into partially overlapping segments and write metadata files.
+//
+//-----------------------------
+void run_program(split_arguments const & arguments)
+{    
+    // Linear scan over reference file to extract metadata
+    reference_metadata reference(arguments.ref_file);
+    reference.to_file(arguments.ref_out);
+
+    // Decide on number of bins and default length of segment (last segment of a reference sequence can have different length)
+    uint32_t segment_len = arguments.min_len; 
+    if (uint32_t(std::ceil(reference.total_len / arguments.min_bins)) <= arguments.min_len)
```

Comment:
Integer division is always floor. This shouldn't work as you thought it will. I think if you want to have a ceil like behaviour here I would do:

```c++
(reference.total_len + arguments.min_bins - 1) / arguments.min_bins;
```

Also one trick here is to use

```
// at least min_len
uint32_t segment_len = std::max(reference.total_len / arguments.min_bins, arguments.min_len);
```

this makes it much more clear that you are limiting (i.e. clamping) the segment_length. Another benefit is that you only do the computation once :)

Reply:
`std::max` is a nice one, thanks!